test_common/harness/featureHelpers.cpp

```diff
@@ -0,0 +1,77 @@
+//
+// Copyright (c) 2020 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "featureHelpers.h"
+#include "errorHelpers.h"
+
+#include <assert.h>
+#include <string.h>
+
+#include <vector>
+
+int get_device_cl_c_features(cl_device_id device, OpenCLCFeatures* features)
+{
+    assert(features != NULL);
```

Comment:
Good point - fixed.

We could also just return the structure by value if we don't care about returning a possible error value, since this function shouldn't be called frequently.